src/Dev/Deprecation.php

```diff
@@ -163,8 +174,53 @@ protected static function get_called_method_from_trace($backtrace, $level = 1)
         if ($level == 4 && ($backtrace[$newLevel]['class'] ?? '') === InjectionCreator::class) {
             $newLevel = $newLevel + 4;
         }
+        // handle noticeWithNoReplacment()
+        foreach ($backtrace as $trace) {
+            if (($trace['class'] ?? '') === Deprecation::class
+                && ($trace['function'] ?? '') === 'noticeWithNoReplacment'
+            ) {
+                $newLevel = $newLevel + 1;
+                break;
+            }
+        }
```

Comment:
Why is this going through a loop when everything else is just checking the original level and the class at either that or the new level? Surely we only care about this method at a specific point (probably right at the top of the trace), and not _literally anywhere it might happen to be anywhere in the trace_?

It's not _likely_ we'll have chained code resulting in calling deprecated code down the line somewhere, but it's not impossible either.

Reply:
There's a bunch of different scenarios to cater for as this logic is shared with both the getting what was called and what the caller was, and this was literally the only solution I found after a couple of hours of trying that actually worked